### PR TITLE
+htp,ben HttpBlueprint benchmark

### PR DESCRIPTION
**Pull request by ktoso**
Raw HttpBlueprint benchmark

cc @johanandren @drewhk

### CODE REVIEW

**Comment by ktoso** [comment]
```
[info] Benchmark                                   Mode  Cnt       Score       Error  Units
[info] HttpBlueprintBenchmark.run_10000_reqs      thrpt   20  132301.546 ± 33270.845  ops/s
[info] HttpBlueprintBenchmark.run_100_reqs        thrpt   20   67225.185 ±  5886.907  ops/s
```

File: akka-bench-jmh/src/main/scala/akka/http/HttpBlueprintBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  def run_100_reqs(blackhole: Blackhole) = {
+    val n = 100
+    val latch = new CountDownLatch(n)
+
+    val replyCountdown = reply map { x =>
+      latch.countDown()
+      blackhole.consume(x)
+      x
+    }
+    server(n).joinMat(replyCountdown)(Keep.right).run()(materializer)
```
Comment (drewhk) [inline akka-bench-jmh/src/main/scala/akka/http/HttpBlueprintBenchmark.scala]: you can prefuse the thing, and use my Sink trick that materializes to a CountDownLatch. Then disable auto-fusing on the materializer. This way fusing effects will not be measured. Probably best to have both cases.

Comment (ktoso) [inline akka-bench-jmh/src/main/scala/akka/http/HttpBlueprintBenchmark.scala]: yeah good point, even prepped `notFusingMaterializer ` but forgot to use it ;)

**Comment by ktoso** [comment]
I'll merge and improve it in the next commit

**Pull request merged**